tagformat/reformat_tags.go

```diff
@@ -33,11 +33,12 @@ func NewTagReformattingMangler(tagName string,
 // Mangle is called for every field in a struct, and returns the value
 // unchanged other than replacing the specified tag.
 func (k *TagReformattingMangler) Mangle(sf reflect.StructField) ([]reflect.StructField, error) {
-	dialsVal := sf.Tag.Get(k.tag)
-	if dialsVal == "" {
-		return []reflect.StructField{sf}, nil
+	nameVal := sf.Tag.Get(k.tag)
+	if nameVal == "" {
+		// there was no name defined, so just fall back to the field name
+		nameVal = sf.Name
 	}
-	splitTagVal, err := k.decodeCasingFunc(dialsVal)
+	splitTagVal, err := k.decodeCasingFunc(nameVal)
```

Comment:
Hmm, this will rarely work as expected.
If a user specifies anything other than `GoCamelCase` as the decoder, it's most likely the case that the split value will be absolute garbage. (or return an error)

Reply:
That's true.  However, it does mean that in the case where there are no `dials` tags, the first argument to `ez.WithFileKeyNaming` is ignored.  I suppose that's fine though.  It does create an interesting "feature" where fields with `dials` tags will use the requested decoder while those without will snap to Go casing but that's probably a good thing.